Library/Homebrew/cask/audit.rb

```diff
@@ -566,24 +588,54 @@ def audit_livecheck_min_os
       return if min_os.blank?
 
       begin
-        min_os_string = MacOSVersion.new(min_os).strip_patch
+        MacOSVersion.new(min_os).strip_patch
       rescue MacOSVersion::Error
-        return
+        nil
       end
+    end
 
-      return if min_os_string <= HOMEBREW_MACOS_OLDEST_ALLOWED
+    def audit_plist_min_os
+      return unless online?
+      return unless cask.livecheckable?
 
-      cask_min_os = cask.depends_on.macos&.version
+      artifacts = cask.artifacts.select do |k|
+        k.is_a?(Artifact::Pkg) || k.is_a?(Artifact::App) || k.is_a?(Artifact::Binary)
+      end
 
-      return if cask_min_os == min_os_string
+      return if artifacts.empty?
 
-      min_os_symbol = if cask_min_os.present?
-        cask_min_os.to_sym.inspect
-      else
-        "no minimum OS version"
+      downloaded_path = download.fetch
+      primary_container = UnpackStrategy.detect(downloaded_path, type: @cask.container&.type, merge_xattrs: true)
+
+      return if primary_container.nil?
+
+      plist_min_os = T.let(nil, T.untyped)
+
+      Dir.mktmpdir do |tmpdir|
+        tmpdir = Pathname(tmpdir)
+        primary_container.extract_nestedly(to: tmpdir, basename: downloaded_path.basename, verbose: false)
+
+        artifacts.each do |artifact|
+          artifact_path = artifact.is_a?(Artifact::Pkg) ? artifact.path : artifact.source
+          path = tmpdir/artifact_path.relative_path_from(cask.staged_path)
+
+          next unless path.exist?
+
+          plist_path = "#{path}/Contents/Info.plist"
+          next unless File.exist?(plist_path)
```

Comment:
Nit: We probably don't need the path check if we're checking for the existence of the file later on.

Reply:
There will probably be a large number of failures, since up until now we've only had an audit for Sparkle. Everything else has eitehr been checking the .plist manually when the cask is added or referring to documentation.